chall02/dmukaliy.c

```diff
@@ -0,0 +1,27 @@
+#include <stdlib.h>
+
+void    put_hex(int num, char **str, int i)
+{
+    char    *hex = "0123456789abcdef";
+    char    *res = *str;
+
+    res[i + 1] = hex[num % 16];
+    num /= 16;
+    res[i] = hex[num % 16];
+}
+
+char    *hv_rgb2hex(int r, int g, int b)
+{
+    char    *res;
+
+    res = (char*)malloc(sizeof(char) * 8);
+    if (!res || r < 0 || g < 0 || b < 0 || r > 255 || g > 255 || b > 255)
```

Comment:
your user input protection should be on the top of the malloc.

and you should also split your user input protection and your malloc protection